core/src/main/java/com/datastrato/gravitino/authorization/UserGroupManager.java

```diff
@@ -87,68 +81,56 @@ public User addUser(String metalake, String name) throws UserAlreadyExistsExcept
     }
   }
 
-  /**
-   * Removes a User.
-   *
-   * @param metalake The Metalake of the User.
-   * @param user THe name of the User.
-   * @return `true` if the User was successfully removed, `false` otherwise.
-   * @throws RuntimeException If removing the User encounters storage issues.
-   */
-  public boolean removeUser(String metalake, String user) {
+  boolean removeUser(String metalake, String user) {
 
     try {
-      AuthorizationUtils.checkMetalakeExists(store, metalake);
-      return store.delete(ofUser(metalake, user), Entity.EntityType.USER);
+      AuthorizationUtils.checkMetalakeExists(metalake);
+      return store.delete(AuthorizationUtils.ofUser(metalake, user), Entity.EntityType.USER);
     } catch (IOException ioe) {
       LOG.error(
           "Removing user {} in the metalake {} failed due to storage issues", user, metalake, ioe);
       throw new RuntimeException(ioe);
     }
   }
 
-  /**
-   * Gets a User.
-   *
-   * @param metalake The Metalake of the User.
-   * @param user The name of the User.
-   * @return The getting User instance.
-   * @throws NoSuchUserException If the User with the given identifier does not exist.
-   * @throws RuntimeException If getting the User encounters storage issues.
-   */
-  public User getUser(String metalake, String user) throws NoSuchUserException {
+  User getUser(String metalake, String user) throws NoSuchUserException {
     try {
-      AuthorizationUtils.checkMetalakeExists(store, metalake);
-      return store.get(ofUser(metalake, user), Entity.EntityType.USER, UserEntity.class);
+      AuthorizationUtils.checkMetalakeExists(metalake);
+      UserEntity entity =
+          store.get(
+              AuthorizationUtils.ofUser(metalake, user), Entity.EntityType.USER, UserEntity.class);
+
+      List<RoleEntity> roleEntities =
+          roleManager.getValidRoles(metalake, entity.roles(), entity.roleIds());
+
+      return UserEntity.builder()
+          .withId(entity.id())
+          .withName(entity.name())
```

Comment:
I found that here you didn't call `withNamespace`, can you add it to unify them all? Also can you build a help method for namespace like you did for NameIdentifier?

Reply:
Done.